textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexInfo.java

```diff
@@ -78,5 +78,98 @@ static RegexInfo anyChar() {
 		regexInfo.emptyable = false;
 		return regexInfo;
 	}
+	
+	/**
+	 * This function simplifies the regexInfo. <br>
+	 * If there the "exact" set gets too large, it will add exact to the query tree,
+	 * and move relevant pieces into prefix and suffix. <br>
+	 * 
+	 * Simplification is performed under three circumstances. <br>
+	 * 1, the size ofexact set is larger than a threshold. <br>
+	 * 2, the minimum length of strings in exact is larger than {@code gramLength+1}, 
+	 *    if we do not set {@code force = true}. <br>
+	 * 3, the minimum length of strings in exact is larger than {@code gramLength}, 
+	 *    if we set {@code force = true}. <br>
+	 * @param force
+	 */
+	RegexInfo simplify(boolean force) {
+		TranslatorUtils.removeDuplicateAffix(exact, false);
+		
+		if ( exact.size() > TranslatorUtils.MAX_EXACT_SIZE ||
+			( TranslatorUtils.minLenOfString(exact) >= GramBooleanQuery.gramLength && force) ||
+			TranslatorUtils.minLenOfString(exact) >= GramBooleanQuery.gramLength + 1){
+			// Add exact to match (query tree)
+			// Transfer information from exact to prefix and suffix
+			match.add(exact);
+			for (String str: exact) {
+				if (str.length() < 3) {
```

Comment:
It is the "default" length of gram... We should replace it with `GramBooleanQuery.gramLength`. (That is why we make `gramLength` `static`.)